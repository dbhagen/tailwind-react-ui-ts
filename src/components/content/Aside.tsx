import React from 'react'
import PropTypes from 'prop-types'

import ContentBlock from './ContentBlock'

const Aside = ({ is, ...rest }) => <ContentBlock is={is} {...rest} />

Aside.propTypes = {
  is: PropTypes.oneOfType([PropTypes.string, PropTypes.func, PropTypes.object]),
}

Aside.defaultProps = {
  is: 'aside',
}

export default Aside
