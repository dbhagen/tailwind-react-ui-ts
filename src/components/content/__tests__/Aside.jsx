import React from 'react'
import { shallow } from 'enzyme'

import Aside from '../Aside'

const setup = (testProps = {}) => {
  const props = Object.assign(testProps)

  const wrapper = shallow(<Aside {...props}>Lorem ipsum...</Aside>)

  return {
    props,
    wrapper,
  }
}

describe('Aside', () => {
  it('renders matching snapshot', () => {
    const { wrapper } = setup()

    expect(wrapper).toMatchSnapshot()
  })
})
