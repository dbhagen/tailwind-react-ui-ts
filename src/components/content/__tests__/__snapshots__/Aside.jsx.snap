// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aside renders matching snapshot 1`] = `
<WithTheme(ContentBlock)
  is="aside"
>
  Lorem ipsum...
</WithTheme(ContentBlock)>
`;
